leshan-core/src/main/java/org/eclipse/leshan/core/util/X509CertUtil.java

```diff
@@ -0,0 +1,310 @@
+    /**
+     * rfc3280#section-4.2.1.7 - GeneralName
+     */
+    public enum GeneralName {
+
+        /**
+         * otherName                       [0]     OtherName
+         */
+        OTHER_NAME(0),
+
+        /**
+         * rfc822Name                      [1]     IA5String
+         */
+        RFC822_NAME(1),
+
+        /**
+         * dNSName                         [2]     IA5String
+         */
+        DNS_NAME(2),
+
+        /**
+         * x400Address                     [3]     ORAddress
+         */
+        X400_ADDRESS(3),
+
+        /**
+         * directoryName                   [4]     Name
+         */
+        DIRECTORY_NAME(4),
+
+        /**
+         * ediPartyName                    [5]     EDIPartyName
+         */
+        EDI_PARTY_NAME(5),
+
+        /**
+         * uniformResourceIdentifier       [6]     IA5String
+         */
+        UNIFORM_RESOURCE_IDENTIFIER(6),
+
+        /**
+         * iPAddress                       [7]     OCTET STRING
+         */
+        IP_ADDRESS(7),
+
+        /**
+         * registeredID                    [8]     OBJECT IDENTIFIER
+         */
+        REGISTERED_ID(8);
+
+        /**
+         * The code value.
+         */
+        public final int value;
+
+        /**
+         * Instantiates a new code with the specified code value.
+         *
+         * @param value the integer value of the code
+         */
+        private GeneralName(final int value) {
+            this.value = value;
+        }
+
+        /**
+         * Converts the specified integer value to a general name code.
+         *
+         * @param value the integer value
+         * @return the general name code
+         * @throws IllegalArgumentException if the integer value does not represent a valid general name code.
+         */
+        public static GeneralName valueOf(final int value) {
+            switch (value) {
+            case 0:
+                return OTHER_NAME;
+            case 1:
+                return RFC822_NAME;
+            case 2:
+                return DNS_NAME;
+            case 3:
+                return X400_ADDRESS;
+            case 4:
+                return DIRECTORY_NAME;
+            case 5:
+                return EDI_PARTY_NAME;
+            case 6:
+                return UNIFORM_RESOURCE_IDENTIFIER;
+            case 7:
+                return IP_ADDRESS;
+            case 8:
+                return REGISTERED_ID;
+            default:
+                throw new IllegalArgumentException(String.format("Unknown GeneralName class code: %d", value));
+            }
+        }
+    }
+
+    /**
+     * Helper for checking if given character is HEX character.
+     *
+     * @param ch Char to test
+     * @return true if HEX char, false otherwise.
+     */
+    private static boolean isHex(char ch) {
+        return (ch >= '0' && ch <= '9' || ch >= 'A' && ch <= 'F' || ch >= 'a' && ch <= 'f');
+    }
+
+    /**
+     * Parses RFC 2253 name string.
```

Comment:
are you referring to : https://tools.ietf.org/html/rfc2253#section-3 ?

Reply:
yes, in simplified value form, will add reference here too.